src/modules/navigator/mission.cpp

```diff
@@ -1925,3 +1928,167 @@ void Mission::publish_navigator_mission_item()
 
 	_navigator_mission_item_pub.publish(navigator_mission_item);
 }
+
+
+void Mission::calculate_mission_checksums()
+{
+	union {
+		uint8_t uint8_part;
+		uint8_t uint16_part;
+		float float_part;
+	} crc_part;
```

Comment:
What I meant was this:

```
union {
    struct Item {
        uint8_t frame;
        uint16_t command;
        uint8_t autocontinue;
        float param1;
        float param2;
        float param3;
        float param4;
        int32_t x;
        int32_t y;
        int32_t z;
    } __packed;
    uint8_t raw[32];
}
```

Then you copy the fields into this structure and run the crc over `raw`.